cmd/buildctl/common/common.go

```diff
@@ -128,3 +115,29 @@ func ParseTemplate(format string) (*template.Template, error) {
 	}
 	return template.New("").Funcs(funcs).Parse(format)
 }
+
+// resolveTLSFilesFromDir scans a TLS directory for known cert/key filenames.
+func resolveTLSFilesFromDir(tlsDir string) (caCert, cert, key string, err error) {
+	oneOf := func(either, or string) (string, error) {
```

Comment:
nit: simpler if just `func(names ...string)`

Reply:
It made the error message more difficult to write because then I'd have to check the number of arguments for the correct pattern of "X, Y, or Z" or just "X" or "X or Y". Since it's only ever called with two parameters, I just did it this way.